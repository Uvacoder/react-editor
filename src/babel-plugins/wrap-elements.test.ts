//@ts-ignore
import babelPluginSyntaxJsx from '@babel/plugin-syntax-jsx'
import { transform } from '@babel/standalone'
import babelPluginWrapElements from './wrap-elements'

test('wraps JSX elements', () => {
  const code = '<div><h1 style={{ margin: 0 }}>Hello world</h1></div>'

  const result = transform(code, {
    plugins: [babelPluginSyntaxJsx, babelPluginWrapElements],
  }).code?.replace(/;$/, '')

  expect(result).toMatchSnapshot()
})
