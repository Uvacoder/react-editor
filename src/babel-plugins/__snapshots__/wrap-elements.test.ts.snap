// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wraps JSX elements 1`] = `
"<Wrapper __location={{
  start: {
    line: 1,
    column: 0
  },
  end: {
    line: 1,
    column: 53
  }
}}><div><Wrapper style={{
      margin: 0
    }} __location={{
      start: {
        line: 1,
        column: 5
      },
      end: {
        line: 1,
        column: 47
      }
    }}><h1>Hello world</h1></Wrapper></div></Wrapper>"
`;
